packages/aws-cdk/lib/notices.ts

```diff
@@ -4,104 +4,231 @@ import * as path from 'path';
 import * as fs from 'fs-extra';
 import * as semver from 'semver';
 import { debug, print } from './logging';
+import { Configuration } from './settings';
 import { ConstructTreeNode, loadTreeFromDir, some } from './tree';
 import { flatMap } from './util';
 import { cdkCacheDir } from './util/directories';
 import { versionNumber } from './version';
 
 const CACHE_FILE_PATH = path.join(cdkCacheDir(), 'notices.json');
 
-export interface DisplayNoticesProps {
+export interface NoticesProps {
   /**
-   * The cloud assembly directory. Usually 'cdk.out'.
+   * A `Configuration` instance holding CDK context and settings.
    */
-  readonly outdir: string;
+  readonly configuration: Configuration;
```

Comment:
Since we are encapsulating this logic into a class, ee should take the opportunity to narrow down the properties it consumes. Instead of a `Configuration` parameter, it's better to add `shouldDisplay?` and and `output?` (and whatever else, if I'm missing anything).

Reply:
> But returning undefined is also disruptive

Its not disruptive because it forces the caller to handle this case - the caller might be disruptive yes, but there's only so far much we can control.

Reply:
I understand the reasoning but i'm not sure its the right way to go in this case. There are configuration context keys that are very specific to notices, such as `acknowledged-issue-numbers`, and I don't think any other part of the code should be aware of them. 

Having `Notices` accept the specific properties, instead of a `Configuration` object opens the door for providing it with the wrong values, which may result in a very subtle bug. 

Unless you strongly object, i'll keep it this way for now and we can revisit if a need arises.